src/coinbase/wallet.ts

```diff
@@ -50,19 +47,48 @@ export class Wallet {
    * @param model - The wallet model object.
    * @param master - The HD master key.
    * @param seed - The seed to use for the Wallet. Expects a 32-byte hexadecimal with no 0x prefix.
-   * @param addressModels - The models of the addresses already registered with the Wallet.
    * @hideconstructor
    */
-  private constructor(
-    model: WalletModel,
-    master: HDKey | undefined,
-    seed: string | undefined,
-    addressModels: AddressModel[] = [],
-  ) {
+  private constructor(model: WalletModel, master: HDKey | undefined, seed: string | undefined) {
     this.model = model;
     this.master = master;
     this.seed = seed;
-    this.addressModels = addressModels;
+  }
+
+  /**
+   * Lists the Wallets belonging to the User.
+   *
+   * @param pageSize - The number of Wallets to return per page. Defaults to 10
+   * @param nextPageToken - The token for the next page of Wallets
+   * @returns An object containing the Wallets and the token for the next page
+   */
+  public static async listWallets(
+    pageSize: number = 10,
+    nextPageToken?: string,
+  ): Promise<{ wallets: Wallet[]; nextPageToken: string }> {
+    const walletList = await Coinbase.apiClients.wallet!.listWallets(
+      pageSize,
+      nextPageToken ? nextPageToken : undefined,
+    );
+    const wallets = await Promise.all(
+      walletList.data.data.map(async wallet => {
+        return await Wallet.init(wallet, "");
```

Comment:
If `Wallet.init` is not `async` do we need a `Promise.all` here / await?

Reply:
oh, this one is missed. fixed! 🙏

Reply:
I'm going to update the tests/types as well.

Reply:
fixed!